pandas/core/aggregation.py

```diff
@@ -738,3 +641,99 @@ def aggregate_multiple_funcs(obj, arg, _axis):
                 "cannot combine transform and aggregation operations"
             ) from err
         return result
+
+
+def agg_dict_like(obj, arg, _axis):
+    is_aggregator = lambda x: isinstance(x, (list, tuple, dict))
+
+    if _axis != 0:  # pragma: no cover
+        raise ValueError("Can only pass dict with axis=0")
+
+    selected_obj = obj._selected_obj
+
+    # if we have a dict of any non-scalars
+    # eg. {'A' : ['mean']}, normalize all to
+    # be list-likes
+    if any(is_aggregator(x) for x in arg.values()):
+        new_arg: Dict[Label, Union[AggFuncTypeBase, List[AggFuncTypeBase]]] = {}
+        for k, v in arg.items():
+            if not isinstance(v, (tuple, list, dict)):
+                new_arg[k] = [v]
+            else:
+                new_arg[k] = v
+
+            # the keys must be in the columns
+            # for ndim=2, or renamers for ndim=1
+
+            # ok for now, but deprecated
+            # {'A': { 'ra': 'mean' }}
+            # {'A': { 'ra': ['mean'] }}
+            # {'ra': ['mean']}
+
+            # not ok
+            # {'ra' : { 'A' : 'mean' }}
+            if isinstance(v, dict):
+                raise SpecificationError("nested renamer is not supported")
+            elif isinstance(selected_obj, ABCSeries):
+                raise SpecificationError("nested renamer is not supported")
+            elif (
+                isinstance(selected_obj, ABCDataFrame) and k not in selected_obj.columns
+            ):
+                raise KeyError(f"Column '{k}' does not exist!")
+
+        arg = new_arg
+
+    else:
+        # deprecation of renaming keys
+        # GH 15931
+        keys = list(arg.keys())
+        if isinstance(selected_obj, ABCDataFrame) and len(
+            selected_obj.columns.intersection(keys)
+        ) != len(keys):
+            cols = sorted(set(keys) - set(selected_obj.columns.intersection(keys)))
+            raise SpecificationError(f"Column(s) {cols} do not exist")
+
+    from pandas.core.reshape.concat import concat
+
+    if selected_obj.ndim == 1:
+        # key only used for output
+        colg = obj._gotitem(obj._selection, ndim=1)
+        results = {key: colg.agg(how) for key, how in arg.items()}
+    else:
+        # key used for column selection and output
+        results = {key: obj._gotitem(key, ndim=1).agg(how) for key, how in arg.items()}
+
+    # set the final keys
+    keys = list(arg.keys())
+
+    # Avoid making two isinstance calls in all and any below
+    is_ndframe = [isinstance(r, ABCNDFrame) for r in results.values()]
+
+    # combine results
+    if all(is_ndframe):
```

Comment:
In my opinion, would be even better to have this if/elif/else statement in a separate function.
I would not suggest the best name, but I can infer from the comment that something containing ``combine`` could be the way to go.
Although, not a big deal.

Reply:
yeah this would be nice, this pass or next ok

Reply:
Agreed, and am working in that direction.